tools/metricsdocs/metricsdocs.go

```diff
@@ -2,43 +2,51 @@ package main
 
 import (
 	"fmt"
-	"sort"
 
-	"github.com/kubevirt/cluster-network-addons-operator/pkg/monitoring"
+	"github.com/machadovilaca/operator-observability/pkg/docs"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/monitoring/metrics"
 	metricsparser "github.com/kubevirt/cluster-network-addons-operator/tools/metrics-parser"
 )
 
-const (
-	opening = `# Cluster Network Addons Operator metrics
-> This file is auto generated by metricsdocs, run 'make generate-doc' in order to update it.
+const tpl = `# Cluster Network Addons Operator Metrics
 
-This document aims to help users that are not familiar with metrics exposed by the Cluster Network Addons Operator.
-All metrics documented here are auto-generated by the utility tool 'tools/metricsdocs' and reflects exactly what is being exposed.
-## Cluster Network Addons Operator Metrics List
-`
+{{- range . }}
 
-	footer = `## Developing new metrics
-After developing new metrics or changing old ones, please run 'make generate-doc' to regenerate this document.`
-)
+{{ $deprecatedVersion := "" -}}
+{{- with index .ExtraFields "DeprecatedVersion" -}}
+    {{- $deprecatedVersion = printf " in %s" . -}}
+{{- end -}}
 
-func main() {
-	metricsList := getMetrics()
-	writeToFile(metricsList)
-}
+{{- $stabilityLevel := "" -}}
+{{- if and (.ExtraFields.StabilityLevel) (ne .ExtraFields.StabilityLevel "STABLE") -}}
+	{{- $stabilityLevel = printf "[%s%s] " .ExtraFields.StabilityLevel $deprecatedVersion -}}
+{{- end -}}
 
-func getMetrics() metricsparser.MetricList {
-	metricsList := metricsparser.ReadFromPrometheusCR()
-	metricsList = metricsparser.MetricsOptsToMetricList(monitoring.MetricsOptsList, metricsList)
+### {{ .Name }}
+{{ print $stabilityLevel }}{{ .Help }}. Type: {{ .Type -}}.
 
-	sort.Slice(metricsList, func(i, j int) bool {
-		return metricsList[i].Name < metricsList[j].Name
-	})
+{{- end }}
 
-	return metricsList
-}
+## Developing new metrics
+
+All metrics documented here are auto-generated and reflect exactly what is being
```

Comment:
Why not mentioned 'make generate-doc' command?

Reply:
I think that command should be mentioned on the CI failing if the metrics are not updated, not on the final doc

Reply:
+1 sure